cpp/src/io/orc/writer_impl.cu

```diff
@@ -507,12 +504,20 @@ orc_streams::orc_stream_offsets orc_streams::compute_offsets(
   size_t rle_data_size = 0;
   for (size_t i = 0; i < streams.size(); ++i) {
     const auto &stream = streams[i];
-    const auto &column = columns[stream.column - 1];
 
-    if (((stream.kind == DICTIONARY_DATA || stream.kind == LENGTH) &&
-         (column.orc_encoding() == DICTIONARY_V2)) ||
-        ((stream.kind == DATA) &&
-         (column.orc_kind() == TypeKind::STRING && column.orc_encoding() == DIRECT_V2))) {
+    auto const is_str_data = [&]() {
+      // First stream is an index stream
+      if (stream.column == 0) return false;
```

Comment:
Might be a good idea to hide this field a little bit.  Someone is going to make this same mistake in the future.

```
struct Stream {
  StreamKind kind = INVALID_STREAM_KIND;
  uint64_t length = 0;   // the number of bytes in the file

  uint32_t column_id() { return col_id; }
  uint32_t column_index() { return col_id - 1; }  <-- maybe throw if col_id == 0
private:
  uint32_t col_id = ~0;  // the column id
};
```

Reply:
Done similarly to the suggestion.